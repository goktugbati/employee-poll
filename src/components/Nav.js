import React from "react";
import { Avatar, Box, Link } from "@mui/material";
import AppBar from "@mui/material/AppBar";
import Toolbar from "@mui/material/Toolbar";
import Button from "@mui/material/Button";
import { connect, useSelector } from "react-redux";
import { logoutAuthedUser } from "../actions/authedUser";
import { Link as RouterLink } from "react-router-dom";

const Nav = ({ dispatch, authedUser, users }) => {
  const state = useSelector((state) => state);

  const stringAvatar = (name) => {
    return {
      sx: {
        bgcolor: "#E59866",
      },
      children: `${name.split(" ")[0][0]}${name.split(" ")[1][0]}`,
    };
  };

  return (
    <div>
      {/*<CssBaseline />*/}
      <AppBar component="nav" sx={{ zIndex: 9999 }}>
        <Toolbar>
          {users[authedUser.id].avatarURL ? (
            <Avatar src={users[authedUser.id].avatarURL} />
          ) : (
            <Avatar {...stringAvatar(authedUser.name)} />
          )}
          <Box sx={{ flexGrow: 1, padding: 1 }}>
            {authedUser && authedUser.id && <div>{authedUser.id}</div>}
          </Box>
          <Box sx={{ flexGrow: 1 }}>
            <h2 data-testid="username">
              {authedUser && authedUser.name && <div>{authedUser.name}</div>}
            </h2>
          </Box>

          <Box sx={{ display: { xs: "none", sm: "block" } }}>
            <Button>
              <Link
                component={RouterLink}
                style={{ color: "#fff" }}
                variant="body1"
                to={"/"}
              >
                Home
              </Link>
            </Button>
            <Button sx={{ color: "#fff" }}>
              <Link
                component={RouterLink}
                style={{ color: "#fff" }}
                variant="body1"
                to={"/leaderboard"}
              >
                Leaderboard
              </Link>
            </Button>
            <Button sx={{ color: "#fff" }}>
              <Link
                component={RouterLink}
                style={{ color: "#fff" }}
                variant="body1"
                to={"/add"}
              >
                New Poll
              </Link>
            </Button>
            <Button>
              <Link
                component={RouterLink}
                sx={{ color: "#fff" }}
                to="/login"
                onClick={() => dispatch(logoutAuthedUser())}
                variant="body1"
              >
                Logout
              </Link>
            </Button>
          </Box>
        </Toolbar>
      </AppBar>
    </div>
  );
};
const mapStateToProps = ({ authedUser, users }) => ({
  authedUser,
  users,
});

export default connect(mapStateToProps)(Nav);
