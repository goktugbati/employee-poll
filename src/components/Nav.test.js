import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { render, screen } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import reducer from "../reducers";
import Nav from "./Nav";

const store = configureStore({
  reducer: reducer,
  preloadedState: {
    authedUser: {
      id: "tylermcginnis",
      password: "abc321",
      name: "Tyler McGinnis",
    },
    users: {
      tylermcginnis: {
        id: "tylermcginnis",
        name: "Tyler McGinnis",
        password: "abc321",
        avatarURL:
          "https://pbs.twimg.com/profile_images/1428205319616798721/xmr7q976_400x400.jpg",
      },
    },
  },
});
describe("Navbar", () => {
  test("should render the component", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });

  test("should display username", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    );
    const userName = screen.getByTestId("username");
    expect(userName.textContent).toBe("Tyler McGinnis");
  });
});
