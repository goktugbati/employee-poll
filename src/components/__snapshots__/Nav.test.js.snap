// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-gg2jnu-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="https://pbs.twimg.com/profile_images/1428205319616798721/xmr7q976_400x400.jpg"
              />
            </div>
            <div
              class="MuiBox-root css-tq4qsl"
            >
              <div>
                tylermcginnis
              </div>
            </div>
            <div
              class="MuiBox-root css-i9gxme"
            >
              <h2
                data-testid="username"
              >
                <div>
                  Tyler McGinnis
                </div>
              </h2>
            </div>
            <div
              class="MuiBox-root css-130f8nx"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <a
                  class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineAlways css-jj9aj1-MuiTypography-root-MuiLink-root"
                  href="/"
                  style="color: rgb(255, 255, 255);"
                >
                  Home
                </a>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <a
                  class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineAlways css-jj9aj1-MuiTypography-root-MuiLink-root"
                  href="/leaderboard"
                  style="color: rgb(255, 255, 255);"
                >
                  Leaderboard
                </a>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <a
                  class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineAlways css-jj9aj1-MuiTypography-root-MuiLink-root"
                  href="/add"
                  style="color: rgb(255, 255, 255);"
                >
                  New Poll
                </a>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <a
                  class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineAlways css-e8ta0z-MuiTypography-root-MuiLink-root"
                  href="/login"
                >
                  Logout
                </a>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-gg2jnu-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://pbs.twimg.com/profile_images/1428205319616798721/xmr7q976_400x400.jpg"
            />
          </div>
          <div
            class="MuiBox-root css-tq4qsl"
          >
            <div>
              tylermcginnis
            </div>
          </div>
          <div
            class="MuiBox-root css-i9gxme"
          >
            <h2
              data-testid="username"
            >
              <div>
                Tyler McGinnis
              </div>
            </h2>
          </div>
          <div
            class="MuiBox-root css-130f8nx"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <a
                class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineAlways css-jj9aj1-MuiTypography-root-MuiLink-root"
                href="/"
                style="color: rgb(255, 255, 255);"
              >
                Home
              </a>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <a
                class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineAlways css-jj9aj1-MuiTypography-root-MuiLink-root"
                href="/leaderboard"
                style="color: rgb(255, 255, 255);"
              >
                Leaderboard
              </a>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <a
                class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineAlways css-jj9aj1-MuiTypography-root-MuiLink-root"
                href="/add"
                style="color: rgb(255, 255, 255);"
              >
                New Poll
              </a>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <a
                class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineAlways css-e8ta0z-MuiTypography-root-MuiLink-root"
                href="/login"
              >
                Logout
              </a>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
